sklearn/metrics/_scorer.py

```diff
@@ -247,7 +247,7 @@ def _score(self, method_caller, clf, X, y, sample_weight=None):
         if y_type == "binary":
             if y_pred.shape[1] == 2:
                 y_pred = y_pred[:, 1]
-            else:
+            elif y_pred.shape[1] == 1:  # not multiclass
```

Comment:
Hmm, why is this useful?

Reply:
Looking at the blame, this was added in https://github.com/scikit-learn/scikit-learn/pull/12486 to resolve https://github.com/scikit-learn/scikit-learn/issues/7598

It looks like it was trying to get a better error message for the `y_pred.shape[1]==1` case.